core/src/main/java/org/hibernate/ogm/dialect/spi/AssociationTypeContext.java

```diff
@@ -20,13 +20,20 @@
 public interface AssociationTypeContext {
 
 	/**
-	 * Provide access to the options set for an entity or property.
+	 * Provide access to the options set for the association.
 	 *
-	 * @return a context object providing access to the options effectively applying for a given entity or property.
+	 * @return a context object providing access to the options effectively applying for the association.
 	 */
 	OptionsContext getOptionsContext();
 
 	/**
+	 * Provide access to the options set for the entity owner of the association.
+	 *
+	 * @return a context object providing access to the options effectively applying for the owner entity.
+	 */
+	OptionsContext getOwnerEntityOptionsContext();
```

Comment:
> But let's define the use case clearly first, and assign it a (low?) priority accordingly.

Yes, I don't think it has a lot to do with this commit. This commit just fixes an existing buggy behavior which can have quite undesirable consequences.